main.rb

```diff
@@ -0,0 +1,142 @@
+# rubocop: disable Metrics/ModuleLength
+# rubocop: disable Metrics/PerceivedComplexity, Metrics/CyclomaticComplexity
+
+module Enumerable
+  def my_each
+    return to_enum :my_each unless block_given?
+
+    index = 0
+    while index < size
+      if is_a? Array
+        yield self[index]
+      elsif is_a? Hash
+        yield keys[index], self[keys[index]]
+      elsif is_a? Range
+        yield to_a[index]
+      end
+      index += 1
+    end
+  end
+
+  def my_each_with_index
+    return to_enum :my_each unless block_given?
+
+    index = 0
+    while index < size
+      if is_a? Array
+        yield self[index], index
+      elsif is_a? Hash
+        yield keys[index], self[keys[index]]
+      elsif is_a? Range
+        yield to_a[index], index
+      end
+      index += 1
+    end
+  end
+
+  def my_select
+    return to_enum :my_select unless block_given?
+
+    if is_a? Array
+      results = []
+      my_each { |x| results << x if yield x }
+    else
+      results = {}
+      my_each { |k, v| results[k] = v if yield k, v }
+    end
+    results
+  end
+
+  def my_any?(args = nil)
+    block_is_true = false
+    if block_given?
+      my_each { |x| block_is_true = true if yield(x) }
+    elsif args.is_a? Regexp
+      my_each { |x| block_is_true = true if x.match(args) }
+    elsif args.is_a? Module
+      my_each { |x| block_is_true = true if x.is_a?(args) }
+    elsif args.nil?
+      my_each { |x| block_is_true = true unless x.nil? || x == false }
+    else
+      my_each { |x| block_is_true = true if x.nil? || x == false }
+    end
+    block_is_true
+  end
```

Comment:
- [ ] should have working `my_any?` identical and returns the same thing as ruby's [`any?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-any-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_any.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

 #### my_any? 

- [ ] when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern

```ruby
[3,4,7,11].my_any?(3) should be equal to this [3,4,7,11].any?(3) => true
```
- Yours retrieves the wrong answer
![image](https://user-images.githubusercontent.com/52209067/78312693-d3692a80-7511-11ea-8a33-76e690cb8690.png)

- Real any?
![image](https://user-images.githubusercontent.com/52209067/77689883-10f51300-6f68-11ea-8a4c-319977bb3842.png)